packages/core/base/src/adapter.ts

```diff
@@ -54,3 +83,16 @@ export enum WalletAdapterNetwork {
     Testnet = 'testnet',
     Devnet = 'devnet',
 }
+
+export function scopePollingDetectionStrategy(performDetection: () => boolean): void {
+    if (typeof window === 'undefined' || typeof document === 'undefined') return;
+    if (document.readyState === 'complete') {
+        performDetection();
+        return;
+    }
+    function listener() {
+        window.removeEventListener('load', listener);
+        performDetection();
+    }
+    window.addEventListener('load', listener);
+}
```

Comment:
Yeah, I think this is good. Why does `performDetection()` return a value?

Reply:
Nice, you noticed!

When we introduce more aggressive polling, the return value will let us know when to _stop_ polling, because the wallet was detected.

Reply:
Ah, good call. I forgot `pollUntilReady` used to do just that.